sdk/core/azure-core/src/main/java/com/azure/core/http/policy/RequestIdPolicy.java

```diff
@@ -3,22 +3,53 @@
 
 package com.azure.core.http.policy;
 
+import com.azure.core.http.HttpHeader;
 import com.azure.core.http.HttpPipelineCallContext;
 import com.azure.core.http.HttpPipelineNextPolicy;
 import com.azure.core.http.HttpResponse;
 import reactor.core.publisher.Mono;
 
+import java.util.Objects;
 import java.util.UUID;
+import java.util.function.Supplier;
 
 /**
  * The pipeline policy that puts a UUID in the request header. Azure uses the request id as
  * the unique identifier for the request.
  */
 public class RequestIdPolicy implements HttpPipelinePolicy {
     private static final String REQUEST_ID_HEADER = "x-ms-client-request-id";
+    private final Supplier<HttpHeader> requestIdSupplier;
+
+
+    /**
+     * Creates a RequestIdPolicy provided {@link Supplier}.
+     *
+     * @param headerSupplier The {@link Supplier} to provide header to add to outgoing requests.
+     */
+    public RequestIdPolicy(Supplier<HttpHeader> headerSupplier) {
```

Comment:
should we make it clear in the java-doc about assumptions?

By default RequestIdPolicy sets `x-ms-client-request-id` header if it is not already set. A different header can be treated as requestId header by returning a `non-null` Header from `requestIdSupplier`. Header returned by the `requestIdSupplier` will be ignored if it is `null` or there exists a header with same name, if `null` then `x-ms-client-request-id` will be set.